include/zephyr/drivers/video-controls.h

```diff
@@ -31,13 +38,21 @@ extern "C" {
 
 /**
  * @name Control classes
+ *
+ * List of control classes, to group related controls together in blocks.
+ *
+ * This list is kept identical to the Linux kernel definitions.
+ *
  * @{
  */
-#define VIDEO_CTRL_CLASS_GENERIC	0x00000000	/**< Generic class controls */
-#define VIDEO_CTRL_CLASS_CAMERA		0x00010000	/**< Camera class controls */
-#define VIDEO_CTRL_CLASS_MPEG		0x00020000	/**< MPEG-compression controls */
-#define VIDEO_CTRL_CLASS_JPEG		0x00030000	/**< JPEG-compression controls */
-#define VIDEO_CTRL_CLASS_VENDOR		0xFFFF0000	/**< Vendor-specific class controls */
+#define VIDEO_CTRL_CLASS_BASE			0x00980900 /**< Base class controls */
+#define VIDEO_CTRL_CLASS_CODEC_BASE		0x00990900 /**< Stateful codec controls */
```

Comment:
@josuah : I don't want to delay merging this PR but because you will need to rebase this PR anyways (due to https://github.com/zephyrproject-rtos/zephyr/pull/76144 merged), I take occasion to add this comment.

I think it's better to rename the control as well to be synced with Linux, for example :

- `VIDEO_CTRL_CLASS_CAMERA_BASE` to `VIDEO_CID_CAMERA_CLASS_BASE`

Reply:
I appreciate the feedback. A lot easier to fix it now than after it is merged!

Reply:
Done. Feel free to ask for more changes as much as desired.